app/models/metric/ci_mixin/processing.rb

```diff
@@ -197,4 +199,23 @@ def normalize_value(value, counter)
     end
     return value, message
   end
+
+  def publish_metrics(metrics)
+    return if MiqQueue.messaging_type == "miq_queue"
+
+    metrics.each_value do |metric|
+      resource = metric.delete(:resource)
+      metric[:resource_type] = resource.class.base_class.name
+      metric[:resource_id]   = resource.id
+      metric[:resource_ref]  = resource.ems_ref if resource.respond_to?(:ems_ref)
+      metric[:resource_uid]  = resource.uid_ems if resource.respond_to?(:uid_ems)
```

Comment:
@agrare Do we need some sort of ems_id here, or is that already in metric?

Reply:
Actually I see it's in the sender, but I'm wondering if that should also be in the payload for completeness

Reply:
That is already in the payload, `"parent_ems_id"=>2,`

Reply:
The `parent` bit is weird, but that's ok for now.